synthesis/fragment/fragment.go

```diff
@@ -178,5 +190,13 @@ func optimizeOverhangIteration(sequence string, minFragmentSize int, maxFragment
 // last 4 base pairs are the initial overhang set.
 func Fragment(sequence string, minFragmentSize int, maxFragmentSize int, existingOverhangs []string) ([]string, float64, error) {
 	sequence = strings.ToUpper(sequence)
-	return optimizeOverhangIteration(sequence, minFragmentSize, maxFragmentSize, []string{}, append([]string{sequence[:4], sequence[len(sequence)-4:]}, existingOverhangs...))
+	return optimizeOverhangIteration(sequence, minFragmentSize, maxFragmentSize, []string{}, append([]string{sequence[:4], sequence[len(sequence)-4:]}, existingOverhangs...), []string{})
+}
+
+// FragmentWithOverhangs fragments a sequence with only a certain overhang set.
+// This is useful if you are constraining the set of possible overhangs when
+// doing more advanced forms of cloning.
+func FragmentWithOverhangs(sequence string, minFragmentSize int, maxFragmentSize int, existingOverhangs []string, buildOverhangs []string) ([]string, float64, error) {
+	sequence = strings.ToUpper(sequence)
```

Comment:
On the flip side, something like #391 is why I like using raw sequence - though I guess I could just do the same with the sequence / methylation sites suggested by #347 , and it probably is a bad idea to have semantic meaning to upper/lowercase. I kind of like having a string because it is so low level - you're pretty directly working on the data structure, instead of having to go through conversions if you're working with sequence from another location.

Overall, I think it's a good idea, but aesthetically I like strings

Reply:
I'd describe the benefit more as having a "funnel" (the constructor) through which all external representations of DNA go through, once it's turned into your internal DNA struct you no longer have to do any conversions, it's always in the format you expect.

A quick google shows that this is what [biopython](https://biopython.org/wiki/Seq) does